src/bin/scheduler/setup/directories.rs

```diff
@@ -368,6 +370,66 @@ fn clean_up_results_directory(
     Ok(results_directory_lock.release()?)
 }
 
+struct OwnershipSetter {
+    #[cfg(unix)]
+    user_id: u32,
+    #[cfg(unix)]
+    group_id: u32,
+    #[cfg(windows)]
+    user: String,
+}
+
+impl OwnershipSetter {
+    pub fn make_for_current_user() -> AnyhowResult<Self> {
+        #[cfg(unix)]
+        {
+            Ok(Self {
+                user_id: unsafe { libc::getuid() },
+                group_id: unsafe { libc::getgid() },
+            })
+        }
+        #[cfg(windows)]
+        {
+            Ok(Self {
+                user: whoami::username(),
+            })
+        }
+    }
+
+    pub fn take_ownership(&self, target: &Utf8Path, recursive: bool) -> AnyhowResult<()> {
```

Comment:
Having two functions here without the bool argument would be cleaner. Maybe the recursive function can call the non-recursive one (to deduplicate the `format!` string)